pkg/apis/application/v1alpha1/application_annotations.go

```diff
@@ -6,10 +6,12 @@ const (
 	AnnotationKeyRefresh string = "argocd.argoproj.io/refresh"
 	// AnnotationKeyHydrate is the annotation key which indicates that app needs to be hydrated. Removed by application controller after app is hydrated.
 	AnnotationKeyHydrate string = "argocd.argoproj.io/hydrate"
-
 	// AnnotationKeyManifestGeneratePaths is an annotation that contains a list of semicolon-separated paths in the
 	// manifests repository that affects the manifest generation. Paths might be either relative or absolute. The
 	// absolute path means an absolute path within the repository and the relative path is relative to the application
 	// source path within the repository.
 	AnnotationKeyManifestGeneratePaths = "argocd.argoproj.io/manifest-generate-paths"
+	// AnnotationKeyTrackingMethod is used to record how Argo CD tracked the resource (label, annotation, or both).
+	// It helps ensure only the tracking metadata added by Argo CD is removed later.
+	AnnotationKeyTrackingMethod = "argocd.argoproj.io/tracking-method"
```

Comment:
We should pass the tracking method as a param on the function instead of passing it on the manifest.